src/paymentservice/src/main/java/ca/bc/gov/open/pssg/rsbc/dps/paymentservice/CalculateSinglePayment.java

```diff
@@ -19,65 +25,106 @@
-   * IS EXPECTED AND WHAT THE LEGACY SYSTEM DOES. PLEASE DO NOT CHANGE THIS. 
-   * 
-   * @param ex
-   * @return
-   */
-  @ExceptionHandler(MissingServletRequestParameterException.class)
-  public SinglePaymentResponse handleMissingParams(MissingServletRequestParameterException ex) {
-      
-	  // TODO - Add appropriate logging here.  
-	  
-	  String paramName = ex.getParameterName();
-      // log.fatal (....
-     
-	 String errMsg = String.format(PaymentServiceConstants.PAYMENT_SERVICE_ERR_MISSING_PARAM, paramName);
-	  
-	 return new SinglePaymentResponse(errMsg, PaymentServiceConstants.PAYMENT_SERVICE_FAILURE_CD, null);
-      
-  }
+	@Value("${bambora.hostedpaymentendpoint}")
+	private String hostedPaymentEndpoint;
+
+	@Value("${bambora.merchantid}")
+	private String merchantId;
+
+	@Value("${bambora.hashkey}")
+	private String hashKey;
+
+	/**
+	 * 
+	 * singlepaymenturl
+	 * 
+	 * WARNING: Do not modify parameters. This operation must replicate legacy
+	 * system exactly.
+	 * 
+	 * Note: In the event of an exception, this methods will return an HTTP status
+	 * code of 200. THIS IS EXPECTED AND WHAT THE LEGACY SYSTEM DOES. PLEASE DO NOT
+	 * CHANGE THIS. An error is indicated to the calling system by the -1 status
+	 * code in the response structure.
+	 * 
+	 * @param transType
+	 * @param invoiceNumber
+	 * @param approvedPage
+	 * @param declinedPage
+	 * @param errorPage
+	 * @param totalItemsAmount
+	 * @param ref1
+	 * @param ref2
+	 * @param ref3
+	 * @param minutesToExpire
+	 * @return
+	 * @throws PaymentServiceException 
+	 */
+	@RequestMapping("/getSinglePaymentURL")
+	public SinglePaymentResponse singlepaymenturl(@RequestParam(value = "transType", required = true) String transType,
+			@RequestParam(value = "invoiceNumber", required = true) String invoiceNumber,
+			@RequestParam(value = "approvedPage", required = true) String approvedPage,
+			@RequestParam(value = "declinedPage", required = true) String declinedPage,
+			@RequestParam(value = "errorPage", required = true) String errorPage,
+			@RequestParam(value = "totalItemsAmount", required = true) String totalItemsAmount,
+			@RequestParam(value = "ref1", required = false) String ref1,
+			@RequestParam(value = "ref2", required = false) String ref2,
+			@RequestParam(value = "ref3", required = false) String ref3,
+			@RequestParam(value = "minutesToExpire", required = true) String minutesToExpire) {
+
+		// TODO - Complete logging once available.
+
+		PaymentClient client = null;
+
+		try {
+			
+			// Make sure autowired params are not null (May be missing from configMap)
+			if ( StringUtils.isEmpty(hostedPaymentEndpoint) || StringUtils.isEmpty(merchantId) || StringUtils.isEmpty(hashKey)) 
+				throw new PaymentServiceException(PaymentServiceConstants.PAYMENT_SERVICE_ERR_MISSING_CONFIG_PARAMS);
+
+			client = new BamboraClientImpl(new URL(hostedPaymentEndpoint), merchantId, hashKey,
```

Comment:
We could inject the client in the constructor of the controller avoiding the new keyword here.

Reply:
Not so easy because the injected values (hostedPaymentEndpoint, merchantId, hashKey)are not available at the time of construction and the minutesToExpire is passed in to the controller provided endpoint at run time not at object instantiation time.

Reply:
ok makes sense